src/icalendar/tests/test_issue_722_generate_vtimezone.py

```diff
@@ -0,0 +1,408 @@
+        ("Europe/Berlin", datetime(2024, 10, 27, 4, 0), "CET"),
+
+        # transition times from https://www.zeitverschiebung.net/de/timezone/america--new_york
+        ("America/New_York", datetime(1970, 1, 1), "EST"),
+        # Daylight Saving Time
+        ("America/New_York", datetime(2024, 11, 3, 0, 0), "EDT"),
+        ("America/New_York", datetime(2024, 11, 3, 1, 0), "EDT"),
+        ("America/New_York", datetime(2024, 11, 3, 1, 59, 59), "EDT"),
+        # 03.11.2024 2:00am -> 1:00am Standard
+        # ("America/New_York", datetime(2024, 11, 3, 2, 0), "EDT"),
+        ("America/New_York", datetime(2024, 11, 3, 2, 0, 1), "EST"),
+        ("America/New_York", datetime(2024, 11, 3, 3, 0), "EST"),
+        ("America/New_York", datetime(2025, 3, 9, 1, 0), "EST"),
+        ("America/New_York", datetime(2025, 3, 9, 1, 59, 59), "EST"),
+        ("America/New_York", datetime(2025, 3, 9, 2, 0), "EST"),
+        # 09.03.2025 2:00am -> 3:00am Daylight Saving Time
+        ("America/New_York", datetime(2025, 3, 9, 3, 0), "EDT"),
+        ("America/New_York", datetime(2025, 3, 9, 3, 1, 1), "EDT"),
+        ("America/New_York", datetime(2025, 3, 9, 4, 0), "EDT"),
+    ]
+)
+def test_check_datetimes_around_transition_times(tzp, tzid, dt, tzname):
+    """We should make sure than the datetimes with the generated timezones
+    work as expected: They have the right UTC offset, dst and tzname.
+    """
+    message = f"{tzid}: {dt} (expected in {tzname})"
+    expected_dt = tzp.localize(dt, tzid)
+    component = Timezone.from_tzinfo(tzp.timezone(tzid))
+    generated_tzinfo = component.to_tz(tzp, lookup_tzid=False)
+    generated_dt = dt.replace(tzinfo=generated_tzinfo)
+    print(generated_tzinfo)
+    if tzp.uses_pytz():
+        # generated_dt = generated_tzinfo.localize(dt)
+        generated_dt = generated_tzinfo.normalize(generated_dt)
+        if dt in (
+                datetime(2024, 10, 27, 1, 0),
+                datetime(2024, 11, 3, 1, 59, 59),
+                datetime(2024, 11, 3, 1, 0),
+                datetime(2024, 11, 3, 0, 0),
+                datetime(2024, 10, 27, 2, 59, 59),
+                datetime(2024, 10, 27, 2, 30),
+                datetime(2024, 10, 27, 2, 0),
+                datetime(2024, 10, 27, 1, 0)
+            ):
+            pytest.skip("We do not know how to do this. PYTZ-TODO")
+    assert generated_dt.tzname() == expected_dt.tzname() == tzname, message
+    assert generated_dt.dst() == expected_dt.dst(), message
+    assert generated_dt.utcoffset() == expected_dt.utcoffset(), message
+
+
+@pytest.mark.parametrize(
+    "uid", [0, 1, 2, 3]
+)
+def test_dateutil_timezone_when_time_is_going_backwards(calendars, tzp, uid):
+    """Sometimes, times can be ambiguous: We go back from 3am to 2 am.
+
+    By the RFC 5545, we cannot accommodate this case. All datetimes should
+    be BEFORE the transition if ambiguous. However, dateutil can
+    create a timezone that allows being after of cause.
```

Comment:
I'm not sure what this sentence was trying to say, so this is my best guess. Please alter as needed.

```suggestion
    create a timezone that allows the event to be after this ambiguous time span, of course.
```